src/tribol/mesh/CouplingScheme.hpp

```diff
@@ -321,11 +321,119 @@ class CouplingScheme
    * \param [in] t simulation time at given cycle
    *
    */
-   void writeInterfaceOutput( const std::string& dir,
+  void writeInterfaceOutput( const std::string& dir,
                               const VisType v_type, 
                               const integer cycle, 
                               const real t );
 
+  /**
+   * @brief Check if coupling scheme has MFEM mesh data
+   * 
+   * @return true: MFEM mesh data exists
+   * @return false: MFEM mesh data does not exist
+   */
+  bool hasMfemData() const { return m_mfemMeshData != nullptr; }
+
+  /**
+   * @brief Get the MFEM mesh data object
+   * 
+   * @return MfemMeshData* 
+   */
+  MfemMeshData* getMfemMeshData() { return m_mfemMeshData.get(); }
+  
+  /**
+   * @brief Get the MFEM mesh data object (const overload)
+   * 
+   * @return const MfemMeshData* 
+   */
+  const MfemMeshData* getMfemMeshData() const
+  {
+    return m_mfemMeshData.get();
+  }
+
+  /**
+   * @brief Sets the MFEM mesh data object
+   * 
+   * @param mfemMeshData Unique pointer to MFEM mesh data
+   */
+  void setMfemMeshData(std::unique_ptr<MfemMeshData> mfemMeshData)
+  {
+    m_mfemMeshData = std::move(mfemMeshData);
+  }
+
+  /**
+   * @brief Check if coupling scheme has MFEM submesh field data
+   * 
+   * @return true: MFEM submesh field data exists
+   * @return false: MFEM submesh field data does not exist
+   */
+  bool hasMfemSubmeshData() const { return m_mfemSubmeshData != nullptr; }
+
+  /**
+   * @brief Get the MFEM submesh field data object
+   * 
+   * @return MfemSubmeshData* 
+   */
+  MfemSubmeshData* getMfemSubmeshData() { return m_mfemSubmeshData.get(); }
+  
+  /**
+   * @brief Get the MFEM submesh field data object (const overload)
+   * 
+   * @return const MfemSubmeshData* 
+   */
+  const MfemSubmeshData* getMfemSubmeshData() const 
+  {
+    return m_mfemSubmeshData.get();
+  }
+
+  /**
+   * @brief Sets the MFEM submesh field data object
+   * 
+   * @param MfemSubmeshData Unique pointer to MFEM submesh field data
+   */
+  void setMfemSubmeshData(std::unique_ptr<MfemSubmeshData> mfemSubmeshData)
+  {
+    m_mfemSubmeshData = std::move(mfemSubmeshData);
+  }
+
+  /**
+   * @brief Check if coupling scheme has MFEM matrix transfer data
+   * 
+   * @return true: MFEM matrix transfer data exists
+   * @return false: MFEM matrix transfer data does not exist
+   */
+  bool hasMfemMatrixData() const { return m_mfemMatrixData != nullptr; }
+  
+  /**
+   * @brief Get the MFEM matrix transfer data object
```

Comment:
Consider elaborating on what matrix transfer means here for clarity. Let's say I'm implementing a new method with this new API, or modifying an existing one, I want to look at the function prototypes and say, "I need to do that." I think "matrix transfer" is a little ambiguous.

Reply:
Comment has been updated